src/utils/providers/ThemeProvider.tsx

```diff
@@ -0,0 +1,33 @@
+import { FC, useEffect, createContext } from "react"
+
+import { useLocalStorage, IThemeProviderProps, ThemeContextType } from "../../components/index"
+
+
+export const ThemeContext = createContext<ThemeContextType | undefined>(undefined)
+
+export const ThemeProvider: FC<IThemeProviderProps> = ({ children }) => {
+    const [theme, setTheme] = useLocalStorage("Theme", getCurrentTheme())
+
+    useEffect(() => {
+        if (theme === "Dark"){
+            document.body.classList.add('dark')
+        } else {
+            document.body.classList.remove('dark')
+        }
+    }, [theme]);
+
+    function getCurrentTheme(){
```

Comment:
It's better to move this function outside of component as it doesn't use any resources from it